doc/manual/source/store/drv.md

```diff
@@ -0,0 +1,284 @@
+   Since it is a derivation, it must be content-addressed
+
+ - There are no other ways a store object would end up in an input closure.
+   The references of a derivation in store object form always come from solely from the inputs of the derivation.
+
+### Deriving Path {#deriving-path-encoding}
+
+- *constant*
+
+  Constant deriving paths are encoded simply as the underlying store path is.
+  Thus, we see that every encoded store path is also a valid encoded (constant) deriving path.
+
+- *output*
+
+  Output deriving paths are encoded by
+
+  - encoding of a store path referring to a derivation
+
+  - a separator (`^` or `!` depending on context)
+
+  - the name of an output
+
+  > **Example**
+  >
+  > ```
+  > /nix/store/lxrn8v5aamkikg6agxwdqd1jz7746wz4-firefox-98.0.2.drv^out
+  > ```
+  >
+  > This parses like so:
+  >
+  > ```
+  > /nix/store/lxrn8v5aamkikg6agxwdqd1jz7746wz4-firefox-98.0.2.drv^out
+  > |------------------------------------------------------------| |-|
+  > store path (usual encoding)                                    output name
+  >                                                           |--|
+  >                                                           note the ".drv"
+  > ```
+
+## Extending the model to be higher-order
+
+**Experimental feature**: [`dynamic-derivations`](@docroot@/development/experimental-features.md#xp-feature-dynamic-derivations)
+
+We can apply the same extension discussed for the abstract model to the concrete model.
+Again, only the data type for Deriving Paths needs to be modified.
+Derivations are the same except for using the new extended deriving path data type.
+
+```idris
+type OutputName = String
+
+data DerivingPath
+  = ConstantPath { storeObj : StorePath }
+  | Output {
+      drv    : DerivingPath, -- Note: changed
+      output : OutputName,
+    }
+```
+
+Now, the `drv` field of `Output` is itself a `DerivingPath` instead of an `StorePath`.
+
+Under this extended model, `DerivingPath`s are thus inductively built up from an `ConstantPath`, contains in 0 or more outer `Output`s.
+
+### Encoding {#deriving-path-encoding}
+
+The encoding is adjusted in a very simplest way, merely displaying the same
+
+> **Example**
+>
+> ```
+> /nix/store/lxrn8v5aamkikg6agxwdqd1jz7746wz4-firefox-98.0.2.drv^foo.drv^bar.drv^out
+> |----------------------------------------------------------------------------| |-|
+> inner deriving path (usual encoding)                                           output name
+> |--------------------------------------------------------------------| |-----|
+> even more inner deriving path (usual encoding)                         output name
+> |------------------------------------------------------------| |-----|
+> innermost constant store path (usual encoding)                 output name
+> ```
+
+## Extra extensions
+
+### `__structuredAttrs`
+
+Historically speaking, most users of Nix made GNU Bash with a script the command run, regardless of what they were doing.
```

Comment:
Here the text some work needs

Reply:
I am just going to delete this, because it is too explanation-y, and there are langauge docs we can move to store docs for this (in a subsequent PR!).